src/models/Function/FunctionService.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import WebSiteManagementClient from "azure-arm-website";
+import { ServiceClientCredentials, WebResource } from "ms-rest";
+import * as request from 'request-promise';
+import { appendExtensionUserAgent, createAzureClient } from "vscode-azureextensionui";
+import * as Constants from "../../constants";
+import { ApisTreeItem } from "../../explorer/ApisTreeItem";
+import { ApiTreeItem } from "../../explorer/ApiTreeItem";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { nRequest, requestUtil, sendRequest } from "../../utils/requestUtil";
+import { signRequest } from "../../utils/signRequest";
+import { IAzureClientInfo } from "../Azure/AzureClientInfo";
+import { IFunctionKeys } from "./Function";
+import { IFunctionContract } from "./IFunctionContract";
+
+export namespace FunctionService {
+    // Get all functions from a function app
+    export async function listAllFunctions(baseUrl: string, credentials: ServiceClientCredentials): Promise<IFunctionContract[]> {
+        const queryUrl = `${baseUrl}/functions?api-version=${Constants.functionAppApiVersion}`;
```

Comment:
make baseUrl a class level prop

Reply:
if that is the case, then we can also add credentials, subscriptionId, resourceGroup, etc to the class, and change the function props in this file, but it also means that thes functions will have more limitations, since we have to know a lot of props to create a FunctionAppService, then we can use the functions in the service